File: server/datasets/controlled_benchmarks/java/multi_student_cohorts/cohort_g1/renamed_38_bst.java
class Node_2 {
    int val_2;
    Node_2 lhs_2, rhs_2;
    public Node_2(int item) {
        val_2 = item;
        lhs_2 = rhs_2 = null;
    }
}
public class Tree_2 {
    public static void main(String[] args) {
        Tree_2 bst = new Tree_2();
        bst.ins_2(50);
        bst.srch_2(50);
    }
    Node_2 rt_2;
    public void ins_2(int val_2) {
        rt_2 = in_rec_2(rt_2, val_2);
    }
    Node_2 in_rec_2(Node_2 rt_2, int val_2) {
        if (rt_2 == null) return new Node_2(val_2);
        if (val_2 < rt_2.val_2) rt_2.lhs_2 = in_rec_2(rt_2.lhs_2, val_2);
        else if (val_2 > rt_2.val_2) rt_2.rhs_2 = in_rec_2(rt_2.rhs_2, val_2);
        return rt_2;
    }
    Node_2 srch_2(int val_2) {
        return sr_rec_2(rt_2, val_2);
    }
    Node_2 sr_rec_2(Node_2 rt_2, int val_2) {
        if (rt_2 == null || rt_2.val_2 == val_2) return rt_2;
        if (rt_2.val_2 < val_2) return sr_rec_2(rt_2.rhs_2, val_2);
        return sr_rec_2(rt_2.lhs_2, val_2);
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_g1/renamed_38_bst.py
class Node_2:
    def __init__(self, val_2):
        self.val_2 = val_2
        self.lhs_2 = None
        self.rhs_2 = None

class Tree_2:
    def __init__(self):
        self.rt_2 = None

    def ins_2(self, val_2):
        if self.rt_2 is None:
            self.rt_2 = Node_2(val_2)
        else:
            self._ins_2_rec(self.rt_2, val_2)

    def _ins_2_rec(self, nd_2, val_2):
        if val_2 < nd_2.val_2:
            if nd_2.lhs_2 is None:
                nd_2.lhs_2 = Node_2(val_2)
            else:
                self._ins_2_rec(nd_2.lhs_2, val_2)
        elif val_2 > nd_2.val_2:
            if nd_2.rhs_2 is None:
                nd_2.rhs_2 = Node_2(val_2)
            else:
                self._ins_2_rec(nd_2.rhs_2, val_2)

    def srch_2(self, val_2):
        return self._srch_2_rec(self.rt_2, val_2)

    def _srch_2_rec(self, nd_2, val_2):
        if nd_2 is None or nd_2.val_2 == val_2:
            return nd_2
        if val_2 < nd_2.val_2:
            return self._srch_2_rec(nd_2.lhs_2, val_2)
        return self._srch_2_rec(nd_2.rhs_2, val_2)
